### ISSUE TITLE
Bug with Single-Item Selection with ComboBox

### ISSUE THREAD

**Issue by modula2019** [unoplatform/uno.extensions#2393]
## Current behavior

I am following the instructions in the documentation below to implement selection in a combobox. I discover that combobox is able to identify item firstly selected and the items above it. The items below the firstly selected item are ignored. I see this as a bug because this is not what it is intended to do. 

https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/Advanced/Selection.html#implement-selection-in-the-peopleapp

## Expected behavior

The expected behaviour should be to identify every selection when it is made.

## How to reproduce it (as minimally and precisely as possible)
[UnoApp1.zip](https://github.com/user-attachments/files/15978464/UnoApp1.zip)

A sample repo is attached here to check the behavior. The ComboBoxPage.xaml is the affected page

## Environment

Nuget Package (s):

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Android
- [ ] WebAssembly
- [x] Windows
- [ ] Skia (WPF)
- [x] Skia (GTK on Linux/macOS/Windows)
- [ ] Skia (Linux Framebuffer)
- [ ] Build tasks

Visual Studio:

- [ ] 2019 (version: )
- [x] 2022 (version: )
- [ ] Visual Studio Code (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

None. Everything needed is already supplied

**Comment by dr1rrb** [comment]
Checked your repro, indeed there is a bug.

It seems that the `ComboBox` is not un-selecting previous item when selection changes so we end to have more than on item selected and your `SelectedPerson` will remain stick to the first selected item.

Putting this issue into our backlog.

<details>
<summary>Workaround</summary>

In the meantime if you want an **ugly** workaround, you can convert your `IListFeed<Person> People` to a `IListState<Person>` and then add this in your constructor:

```csharp
Task.Run(async () =>
{
    var previousSelection = SelectionInfo.Empty;
    var ctx = SourceContext.GetOrCreate(this);
    await foreach (var _ in ctx.GetOrCreateSource(People).WithCancellation(ctx.Token))
    {
        await People.UpdateMessageAsync(EnforceSingleSelection, ctx.Token);
    }

    void EnforceSingleSelection(MessageBuilder<IImmutableList<Person>> msg)
    {
        var selection = msg.GetSelectionInfo() ?? SelectionInfo.Empty;
        if (selection is { Count: > 1 })
        {
            foreach (var range in previousSelection.Ranges)
            {
                selection = selection.Remove(range);
            }
            global::System.Diagnostics.Debug.Assert(selection.Count == 1);
            msg.Selected(selection);
        }
        previousSelection = selection;
    }
});
```

In short: Each time the `People` state is updated, we will validate that we have more than one item selected, and if so re-update the `People` state to keep only the freshly added item in the selection.

</details>

**Comment by modula2019** [comment]
Thanks for the reply. Your suggestion works while we wait for permanent solution

**Comment by vatsashah45** [comment]
ComboBox doesn't support ISelectionInfo on Windows.
The team needs to fix the support of ICollectionView on Uno Platform Apps.

[ComboTest.zip](https://github.com/user-attachments/files/16144326/ComboTest.zip)

@modula2019 @dr1rrb

**Comment by dr1rrb** [comment]
Hey @modula2019, the workaround I provided was an evolution of my investigation to understand what was the issue. It has the advantage to not alter the code of your app and can be just removed once the bug is going to be fixed. But a simpler workaround would be to remove the `.Selection(SelectedPerson)` from the `IListState<Person> People` and then add a binding on the `ComboBox.SelectedItem={Binding SelectedPerson.Value, Mode=TwoWay}`.

> Note:
> The `.Value` is needed to be able to set an instance of `People` as for the binding the entity is exposed de-normalized to be able to edit each property of the `Person` individually, like `<TextBox Text="{SelectedPerson.FirstName, Mode=TwoWay}" />`.

**Comment by modula2019** [comment]
> Hey @modula2019, the workaround I provided was an evolution of my investigation to understand what was the issue. It has the advantage to not alter the code of your app and can be just removed once the bug is going to be fixed. But a simpler workaround would be to remove the `.Selection(SelectedPerson)` from the `IListState<Person> People` and then add a binding on the `ComboBox.SelectedItem={Binding SelectedPerson.Value, Mode=TwoWay}`.
> 
> > Note:
> > The `.Value` is needed to be able to set an instance of `People` as for the binding the entity is exposed de-normalized to be able to edit each property of the `Person` individually, like `<TextBox Text="{SelectedPerson.FirstName, Mode=TwoWay}" />`.

@dr1rrb Thanks, this makes it much easier. The workaround worked perfectly. Thanks once again

**Comment by agneszitte** [comment]
https://github.com/unoplatform/uno/pull/17582 needs to be done as well to be able to close this issue
